### PR TITLE
Refactor codebase

### PR DESCRIPTION
**Pull request by nicupop729**
## This feature branch holds the following:

- [X] Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- [X] Follow SOLID principles.
- [X] Extract some pieces of logic into separate classes.
- [X] main.rb is only to start the code and ask the user for the initial choice of options (or to quit).

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -4,42 +4,48 @@
 require './book'
 require './rental'
 
-class Main
-  def initialize
-    @people = []
-    @books = []
-    @rentals = []
-  end
+class List
```
Comment (TedLivist) [inline main.rb]: - [x] In order to obey the Single-responsibility rule, the sole purpose of the `main.rb` file should be to start your app, and that is not the case at the moment. Now, let's clean it up a bit. Kindly extract this **List** class into a different file in keeping with the Single-responsibility principle.

File: main.rb [hunk 2]
```diff
@@ -4,42 +4,48 @@
...
+    else
+      @list.each_with_index { |item, i| puts "#{i + 1}: Title: #{item.title}, Author: #{item.author}" }
     end
   end
+end
 
-  def list_books
-    @books.each_with_index { |book, i| puts "#{i + 1}: Title: #{book.title}, Author: #{book.author}" }
-    puts
+class Books
```
Comment (TedLivist) [inline main.rb]: - [x] Kindly extract this **Books** class into a different file in keeping with the SOLID rule.

File: main.rb [hunk 3]
```diff
@@ -4,42 +4,48 @@
...
+    @books = []
   end
 
-  def list_people
-    @people.each_with_index { |person, i| puts "#{i + 1}: Name: #{person.name}, Age: #{person.age} ID: #{person.id}" }
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
```
Comment (TedLivist) [inline main.rb]: - [x] In your file where the Books class has been extracted, in keeping with the KISS and Single-responsibility rules, kindly extract these requests for user input into a different method.

  **EXAMPLE**: You can have that method return an array of your variables, and then destructure your variables in this `create_book` method (see example images below).

  nicu-books    nicu-create

File: main.rb [hunk 4]
```diff
@@ -4,42 +4,48 @@
...
+    new_book = Book.new(title, author)
+    @books << new_book
+    puts 'New book created successfully'
     puts
   end
+end
+
+class People
```
Comment (TedLivist) [inline main.rb]: - [x] Kindly extract this **People** class into a different file.
- [x] In the People class, your methods `create_student` and `create_teacher` have repetitive implementation for user inputs requests. This breaks the DRY rule. You can refactor these requests into one method and call that method where needed.

  nicu-create-stud-teach

File: main.rb [hunk 5]
```diff
@@ -79,38 +85,31 @@ def create_teacher
...
-  def create_book
-    print 'Title: '
-    title = gets.chomp
-    print 'Author: '
-    author = gets.chomp
-    new_book = Book.new(title, author)
-    @books << new_book
-    puts 'New book created successfully'
-    puts
+class Rentals
```
Comment (TedLivist) [inline main.rb]: - [x] Kindly move this **Rentals** class into a different file in keeping with the SOLID rule

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @nicupop729 and @Lino09,

Good job so far!
✔️ Good documentation
✔️ App works
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Nicu, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**